pytest/tests/sanity/state_parts_dump_check.py

```diff
@@ -0,0 +1,115 @@
+NUM_SHARDS = 4
+
+
+def get_dump_check_metrics(target):
+    metrics = target.get_metrics()
+    metrics = [line.split(' ')
+               for line in str(metrics).strip().split('\\n')
+               if re.search('state_sync_dump_check.*{', line)]
+    metrics = {metric: int(val) for metric, val in metrics}
+    return metrics
+
+
+def main():
+    node_config_dump, node_config = state_sync_lib.get_state_sync_configs_pair(
+    )
+    config = load_config()
+    near_root, node_dirs = init_cluster(1, 2, 4, config,
+                                        [["epoch_length", EPOCH_LENGTH]], {
+                                            0: node_config,
+                                            1: node_config_dump,
+                                            2: node_config
+                                        })
+
+    boot_node = spin_up_node(config, near_root, node_dirs[0], 0)
+    logger.info('Started boot_node')
+    dump_node = spin_up_node(config,
+                             near_root,
+                             node_dirs[1],
+                             1,
+                             boot_node=boot_node)
+    dump_check = spin_up_node(config,
+                              near_root,
+                              node_dirs[2],
+                              2,
+                              boot_node=boot_node)
+    dump_check.kill()
+    chain_id = boot_node.get_status()['chain_id']
+    dump_folder = node_config_dump["state_sync"]["dump"]["location"]["Filesystem"]["root_dir"]
+    rpc_address, rpc_port = boot_node.rpc_addr()
+    local_rpc_address, local_rpc_port = dump_check.rpc_addr()
+    cmd = dump_check.get_command_for_subprogram(
+        ('state-parts-dump-check',
+         '--chain-id',
+         chain_id,
+         '--root-dir',
+         dump_folder,
+         'loop-check',
+         '--rpc-server-addr',
+         f"http://{rpc_address}:{rpc_port}",
+         '--prometheus-addr',
+         f"{local_rpc_address}:{local_rpc_port}",
+         '--interval',
+         '2'),
+        dump_check.near_root,
+        dump_check.node_dir,
+        dump_check.binary_name)
+    dump_check.run_cmd(cmd=cmd)
+
+    logger.info('Started nodes')
+
+    wait_for_blocks(node=boot_node, target=EPOCH_LENGTH - 5)
+    # kill dumper node so that it can't dump state.
+    node_height = dump_node.get_latest_block().height
+    logger.info(f'Dump_node is @{node_height}')
+    dump_node.kill()
+    logger.info(f'Killed dump_node')
+
+    wait_for_blocks(node=boot_node, target=EPOCH_LENGTH + 5)
+    # Check the dumped stats.
+    metrics = get_dump_check_metrics(dump_check)
+    assert sum([val for metric, val in metrics.items() if 'state_sync_dump_check_process_is_up' in metric]
+               ) == NUM_SHARDS, f"Dumper process missing for some shards. {metrics}"
+    assert sum([val for metric, val in metrics.items(
+    ) if 'state_sync_dump_check_num_parts_dumped' in metric]) == 0, "No node was supposed to dump parts."
```

Comment:
what I meant here with adding some info after the comma with the assert was actually adding the `metrics` object as is done in the one above. Adding this comment is nice for debugging, but what's actually more valuable is printing out the object that didn't meet our requirements. Otherwise when this line fails we will see that it failed but won't know what `metrics` looked like in order to make this fail

same for all the asserts below